### DISCUSSION TITLE
Should I expect the V8 Error.prepareStackTrace API to work?

### DISCUSSION THREAD

**Discussion by mattwynne** [TypeStrong/ts-node#1517] [Q&A] [answered]
Hi,

We've discovered that, when we try to use the V8 `prepareStackTrace` API to get the structured `CallSite` objects in a stack trace instead of the default string from `new Error().stack`, the line/column numbers are out of whack (though the filenames are correct).

Here's a minimal reproducible example: https://github.com/mattwynne/stacktrace-test

I'm not clear whether this `prepareStackTrace` API is in scope for ts-node, but I wanted to ask whether you were aware of this, and whether there's anything we could do to fix it, or whether there's another way we should be trying to get hold of structured `CallSite` objects for a stack trace.

**Comment by cspotcode** [comment, marked as answer]
We install `@cspotcode/source-map-support` at startup, which implements its own `Error.prepareStackTrace` to rewrite the stack trace strings, apply sourcemaps.  I'm guessing that, when you replace our hook with your own, you're losing the benefit of these mappings.

I believe that the `wrapCallSite` export of `@cspotcode/source-map-support` is what you want to use.  If I recall correctly, it is used to modify each callsite object, applying sourcemaps.  So you can use it to get accurate, source-mapped callsites.

**Reply by mattwynne** [reply]
Yeah, I saw that call to `wrapCallSite` in the source of `Error.prepareStrackTrace` but I hadn't figured out where it was yet. Thanks so much for the prompt response, I'll give that a shot!

**Reply by mattwynne** [reply]
Yes, that's fixed it! ❤️❤️❤️

Big thanks!
